### ISSUE TITLE
Address book example

### ISSUE THREAD

**Issue by egarim** [dreamfactorysoftware/.net-sdk#7]
Hi can you please provide more information about the development method you used to create the address book example it looks like a mix of MVVM and asp.net MVC.

I want to migrate a system that I already have running using DevExpress xaf to dream factory and asp.net I just trying to figure out the a good pattern to use in the migration

**Comment by makosovic** [comment]
Actually its pure MVC pattern. What might confuse you is the ViewModel suffix but it is not the same as in MVVM pattern. M in MVC is the object that is passed to the view, in larger projects it might be a good idea for that M to be a flat object mapped from an entity or multiple entities pulled from db, however in a simple project like this I opted for wrappers around those entities, so I don't have to introduce another complexity of mapping entities to view models and vice versa. 

To sum up, although it might be confusing because of the ViewModel in the name it has nothing to do with MVVM, it's more of a convention and preference how to name your models/view models but the idea stays the same - controller loads the data (to entities or DTO) - creates a model that view knows how to consume (in case of AddressBook somethingViewModel) and passes it to the View.

Let me know if you have any other questions.

Regards
Marko

**Comment by egarim** [comment]
Thanks for the info Marko, well the other question I have is regarding the login and authentication, I have not take a deeper look to it but I guess that's all dream factory. does the SDK provide a mechanism to handle the life of the session?

**Comment by makosovic** [comment]
Actually it's not all dream factory. Life of the session is handled by dream factory, you can login to get sessionId and logout to end the session. However there is also authentication cookie on ASP.NET end which is set and removed in the AccountController.

**Comment by egarim** [comment]
Thanks Marko today I'm going to create a simple MVC app and let you knwo my results
